internal/cosesign1/create.go

```diff
@@ -0,0 +1,134 @@
+package cosesign1
+
+import (
+	"crypto/rand"
+
+	"crypto"
+	"crypto/x509"
+	"encoding/pem"
+	"io"
+	"log"
+
+	"github.com/veraison/go-cose"
+)
+
+func pem2der(chainPem []byte) []byte {
+	block, rest := pem.Decode(chainPem)
+	var r = []byte{}
+	if block.Bytes != nil {
```

Comment:
you also need to check that `block` itself isn't `nil` here and inside the loop below

Reply:
ok, good catch.